src/rdkafka_broker.c

```diff
@@ -5238,20 +5267,36 @@ int rd_kafka_brokers_add0(rd_kafka_t *rk, const char *brokerlist) {
                         break;
 
                 rd_kafka_wrlock(rk);
+                if (rk->rk_conf.enable_bootstrap_servers_canonical_resolve) {
+                        rd_kafka_dbg(rk, ALL, "INIT",
+                                     "Canonicalizing bootstrap broker %s:%d",
+                                     host, port);
+                        sockaddrList = rd_getaddrinfo(
+                            host, RD_KAFKA_PORT_STR, AI_ADDRCONFIG,
```

Comment:
Don't we need to use `AI_CANONNAME` flag here? I think we are only concerned about canonical host name. Other should be discarded. https://man7.org/linux/man-pages/man3/getaddrinfo.3.html